indico/modules/rb_new/client/js/components/modals/BlockingModal.jsx

```diff
@@ -44,124 +51,203 @@ function validate({period, reason, rooms}) {
     return errors;
 }
 
-const formDecorator = createDecorator({
-    field: 'period',
-    updates: (periodValue) => {
-        return {
-            start_date: periodValue[0],
-            end_date: periodValue[1]
-        };
-    }
-}, {
-    field: 'rooms',
-    updates: (rooms) => {
-        return {
-            room_ids: rooms.map((room) => room.id)
-        };
-    }
-}, {
-    field: 'allowed',
-    updates: (allowed) => {
-        return {
-            allowed_principals: allowed.map((obj) => ({
-                id: obj.id,
-                is_group: obj.is_group,
-                provider: obj.provider
-            }))
-        };
-    }
-});
-
-
 class BlockingModal extends React.Component {
     static propTypes = {
-        open: PropTypes.bool.isRequired,
         onClose: PropTypes.func.isRequired,
-        rooms: PropTypes.array,
-        createBlocking: PropTypes.func.isRequired
+        createBlocking: PropTypes.func.isRequired,
+        updateBlocking: PropTypes.func.isRequired,
+        fetchBlockings: PropTypes.func.isRequired,
+        user: PropTypes.object.isRequired,
+        open: PropTypes.bool,
+        mode: PropTypes.oneOf(['read', 'edit', 'new']),
+        blocking: PropTypes.exact({
+            id: PropTypes.number,
+            blocked_rooms: PropTypes.array,
+            allowed: PropTypes.array,
+            start_date: PropTypes.string,
+            end_date: PropTypes.string,
+            reason: PropTypes.string,
+            created_by_id: PropTypes.number
+        })
     };
 
     static defaultProps = {
-        rooms: []
+        open: false,
+        mode: 'new',
+        blocking: {
+            blockingId: null,
+            blocked_rooms: [],
+            allowed: [],
+            start_date: null,
+            end_date: null,
+            reason: ''
+        }
     };
 
-    createBlocking = async (formData) => {
-        const {createBlocking} = this.props;
-        const rv = await createBlocking(formData);
+    constructor(props) {
+        super(props);
+
+        const {mode} = this.props;
+        this.state = {
+            mode
+        };
+    }
+
+    processBlocking = async (formData) => {
+        const {createBlocking, updateBlocking, fetchBlockings, blocking: {id}} = this.props;
+        const {mode} = this.state;
+        let rv;
+
+        if (mode === 'new') {
+            rv = await createBlocking(formData);
+        } else if (mode === 'edit') {
+            rv = await updateBlocking({...formData, blockingId: id});
+            if (!rv.error) {
+                fetchBlockings();
```

Comment:
Do we do some grouping on the server side? IMHO the API call to get the blockings should simply return a list of all the blockings - and in that case, adding the new blocking to that list after creating it will result in the same state as you would get after fetching all theblockings.